File: Python/temp.py

# Lab 12 adding it to Lab 13 lets see how it goes dripped out one 
class TemperatureConverter:
    def __init__(self):
        pass

    def user_greeting(self):
        print("Hello welcome!")
        print("\nThis program will convert degrees Celsius to degrees Fahrenheit.")

    def ask_again(self):
        while True:
            print("\nWould you like to run this program again:")
            print("------------------------------------------")
            user_input = input("If so, please enter either [y/n]: ")
        
            if user_input == 'y':
                return True
            elif user_input == 'n':
                print("----------------------------------------")
                print("Ok then, have a nice day!")
                return False
            else:
                print("-----------------------------------------")
                print("Did not compute, please try again!")

    def convert_and_display(self):
        # Get user input and assign to a variable (float data type)
        celsius = float(input("\nPlease enter the temperature in degrees Celsius; you may use decimals: "))

        # Perform the temperature conversion and assign the result to a variable
        fahrenheit = (celsius * 9/5) + 32

        # Display the results with one decimal point
        print("\n {:.1f} degrees Celsius is equal to {:.1f} degrees Fahrenheit.".format(celsius, fahrenheit))

    def main(self):
        self.user_greeting()

        while True:
            self.convert_and_display()

            # Goes through what happens in the ask_again function to determine the status of the main program
            if not self.ask_again():
                break

File: C++/temp.cpp
#include <iostream>
#include <iomanip>
#include "temperature_converter.h"

class TemperatureConverter{
    // ok the public statement tells C++ that functions outside the public namespace can use what is inside of the namespace
    public:
    void user_greeting(){
        std::cout << "Hello! and welcome to this program!";
        std::cout << "\nThis Program will convert Celsuis to degrees Fahrenheit\n";
    }
    // Since we are asking to return a true a false we use thje boolean data type
    bool askAgain(){
        while (true){
            std::cout << "\nWould you like to run this program again:\n";
            std::cout << "-------------------------------------------";
            char  userInput;
            std::cout << "\nif so either enter a [Y/N]: ";
            std::cin >> userInput; 

            if (userInput == 'y' || userInput == 'Y'){
                return true;
            } else if (userInput == 'n' || userInput == 'N'){
                std::cout << "-------------------------------------------";
                std::cout << "\nok then have a nice day!\n";
                return false;
            } else {
                std::cout << "-------------------------------------------";
                std::cout << "\nDid not compute, please try again\n";
            }
        }

    }

    void convertAndDisplay(){
        // Get user input and assign to a variable (Float data type)
        double celsius;
        std::cout << "\nPlease enter the temperature in degrees celsius:  ";
        std::cin >> celsius; 

        // Performs the temperature in degrees celsius and assigns the result to the decimal point
        double fahrenheit = (celsius * 9/5)+32;

        //Displays the results of the math 
        std::cout << "\n" << std::fixed << std::setprecision(1) << celsius
                  << "\ndegrees Celsius is equal to\n" << fahrenheit << "\ndegrees farenheit.\n";
    }

    void main(){
        user_greeting();

        while (true){
            convertAndDisplay();

            //goes through what happens in the askAgain function
            if (!askAgain()){
                break;
            }
        }
    }
};

// Main  function where all the code happens the main part of the code
// in which the other functions are the C++ versions of the def functions 
int main(){
    TemperatureConverter temperatureConverter;
    temperatureConverter.main();
    return 0;
    
}